We first begin with importing libraries and download csv file that contains the data

In [5]:
# import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
# import data
insurance = pd.read_csv("https://raw.githubusercontent.com/JovianML/opendatasets/master/data/medical-charges.csv")

We first begin with looking into data and look out if the the data is clean nad formatted right.

In [6]:
# Display the first few rows of the dataset
insurance.head()


age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

In [7]:
#check for coluumns types
insurance.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


All the columns look good and are formatted correctly. We can see that age, bmi, children, charges are numbers whereas sex, smoker, and region are possibly categories. 

In [18]:
# Display summary statistics for the numerical columns in the insurance dataset
insurance.describe()

age          bmi     children       charges
count  1337.000000  1337.000000  1337.000000   1337.000000
mean     39.222139    30.663452     1.095737  13279.121487
std      14.044333     6.100468     1.205571  12110.359656
min      18.000000    15.960000     0.000000   1121.873900
25%      27.000000    26.290000     0.000000   4746.344000
50%      39.000000    30.400000     1.000000   9386.161300
75%      51.000000    34.700000     2.000000  16657.717450
max      64.000000    53.130000     5.000000  63770.428010

We can see that the mean values for age, BMI, and children are reasonable. The minimum age is 18 and the maximum is 64, which indicates the age range allowed by the company for insurance. The BMI values also look reasonable, with a minimum of 16 and a maximum of 53, which covers the typical range from underweight to severely obese. The average number of children is about 1.1, which is typical for western countries. The lowest insurance charge is 1122, likely for a customer who rarely visits the doctor, while the highest charge is 63770.

In [ ]:
#check for duplicat values
insurance.drop_duplicates(inplace=True)


0

In [ ]:
insurance.shape

(1337, 7)

Exploratory Analysis and Visualization

In this section, we will explore the dataset by visualizing the distribution of values in key columns. We will also examine the relationships between "charges" and other variables to gain insights into the factors that may influence insurance costs.

for the visualization we use Matplotlib.plotly, Seaborn and plotly

In [3]:
import plotly.express as px
import matplotlib as mat


Let's improve the default style and font sizes for our charts

In [9]:
sns.set_style('darkgrid')
mat.rcParams['font.size'] = 14
mat.rcParams['figure.figsize'] = (10, 6)
mat.rcParams['figure.facecolor'] = '#000000'



Age 

Age is quantitive  column. The minumium age in the dataset is 18 and the maximium is 64. Whe can create for each year a bin using histgram so that we can see the distrubition visually. Also we are going to add a box plot.


In [11]:
insurance.age.describe()

count    1338.000000
mean       39.207025
std        14.049960
min        18.000000
25%        27.000000
50%        39.000000
75%        51.000000
max        64.000000
Name: age, dtype: float64

In [14]:
fig = px.histogram(insurance, x='age', nbins = 47, 
                  title = 'Age Distribution of health Insurance Customers', 
                  marginal = 'box'
                 )
fig.update_layout(bargap = 0.1)
fig.show()

lets find out why there is a lot of 18 - 19 year customers compare to other ages.
first lets look could this be coused by some offers that they are getting.
i want to create I scatter plot between age and charge to see do they have less charges to pay.

In [22]:
age_charge = insurance[['age', 'charges']].copy()
age_charge['charges'] = age_charge['charges'].astype(int)
age_vs_charges_scatter = px.scatter(age_charge, x = 'age',
                                     y = 'charges', 
                                     title = 'Age vs Charges', 
                                     trendline='ols',
                                     trendline_color_override='red')
age_vs_charges_scatter.update_traces(marker=dict(size=10, opacity=0.5, line=dict(width=1, color='DarkSlateGrey')))
age_vs_charges_scatter.update_layout(title_font=dict(size=20), xaxis_title_font=dict(size=16), yaxis_title_font=dict(size=16))
age_vs_charges_scatter.update_xaxes(title_text='Age', title_font=dict(size=16))
age_vs_charges_scatter.update_yaxes(title_text='Charges', title_font=dict(size=16))
age_vs_charges_scatter.update_layout(template='plotly_dark')

age_vs_charges_scatter.show()